miso-web/src/main/webapp/scripts/lims.js

```diff
@@ -79,27 +79,85 @@ var Utils = Utils || {
   },
   createToggleColumn: function(nameOfList) {
     var list = eval(nameOfList);
+    var className = nameOfList.replace(/\./g, '')
+    Utils.registerShiftPressedListener(list) // Used to detect range selection
     return {
         "sTitle": "",
         "mData": "id",
         "mRender": function (data, type, full) {
           var checked = list.indexOf(data) > -1;
 
-          return "<input type=\"checkbox\" onchange=\"Utils.toggleListItem(this.checked, " + data + ", " + nameOfList + ")\"" + (checked ? " checked=\"checked\"" : "") + ">";
+          return "<input id=\"" + data + "_" + className + "\" class=\"" + className + "\" elementId=\"" + data + "\" type=\"checkbox\" onchange=\"Utils.toggleListRange(this.checked, " + data + ", " + nameOfList + ", '" + className + "')\"" + (checked ? " checked=\"checked\"" : "") + ">";
         }
     };
   },
 
+  toggleListRange: function(state, id, list, className) {
+	if(!list.shiftKey) {
+		if(state) {
+		   list.lastSingleSelect = id + "_" + className // Record last click for range selection
+		}
+		Utils.toggleListItem(state, id, list)
+	} else {
+		var currentShiftSelect = id + "_" + className
+		var checkboxes = []
+		jQuery("input." + className ).each(function(){ checkboxes.push(jQuery(this));});
+		
+		function selectRange(selectIndex, shiftIndex) {		
+		    var minSelectIndex = Math.min(selectIndex, shiftIndex)
+		    var maxSelectIndex = Math.max(selectIndex, shiftIndex)
+			checkboxes.forEach(function(element, index) {
+				var elementId = Number(element.attr('elementid'))
+				if(index >= minSelectIndex && index <= maxSelectIndex) {
```

Comment:
This mean that if I check A, then check C, then shift-check E, A will be unchecked. I don't think that's desirable.